src/User/UserRepository.php

```diff
@@ -32,13 +33,122 @@ public function fetchAll()
                 ->setId($row->id)
                 ->setFirstname($row->firstname)
                 ->setLastname($row->lastname)
-                ->setBirthday(new \DateTimeImmutable($row->birthday));
+                ->setBirthday(new DateTimeImmutable($row->birthday))
+                ->setPseudo($row->pseudo)
+                ->setMail($row->mail)
+                ->setPassword($row->password);
 
             $users[] = $user;
         }
 
         return $users;
     }
 
+    public function createUser(User $newUser)
+    {
+        $query = $this->connection->prepare('INSERT INTO "user"(firstname, lastname, birthday, pseudo, mail, password) VALUES (:firstname , :lastname, :birthday, :pseudo, :mail, :password)');
+        $query->bindValue(':firstname', $newUser->getFirstname());
+        $query->bindValue(':lastname', $newUser->getLastname());
+        $query->bindValue(':birthday', $newUser->getBirthday()->format("Y-m-d"));
+        $query->bindValue(':pseudo', $newUser->getPseudo());
+        $query->bindValue(':mail', $newUser->getMail());
+        $query->bindValue(':password', crypt($newUser->getPassword(), 'st'));
+
+        if ($this->pseudoAlreadyExist($newUser->getPseudo()))
+        {
+            return false;
+        }
+
+        $result = $query->execute();
+        if ($result == false)
+        {
+            $query->errorInfo();
+        }
+        return true;
+    }
+
+    public function pseudoAlreadyExist(string $pseudo)
+    {
+        $query = $this->connection->prepare('SELECT pseudo, password FROM "user" WHERE pseudo = :pseudo');
+        $query->bindValue(':pseudo', $pseudo);
+        $query->execute();
+        $rows = $query->fetchAll(PDO::FETCH_OBJ);
+        if (count($rows) > 0)
+        {
+            return true;
+        }
+        return false;
+    }
+
+    public function isLogged()
+    {
+        if (isset($_COOKIE['pseudo']))
+        {
+           return false;
+        }
+        return true;
+    }
+
+    public function rememberUser(int $id, string $pseudo)
+    {
+        if(!isset($_COOKIE['pseudo']))
+        {
+            setcookie('pseudo', $pseudo, time() + 10 * 24 * 3600, null, null, false, true); // 10 jours
+            setcookie('id', strval($id), time() + 10 * 24 * 3600, null, null, false, true); // 10 jours
```

Comment:
```suggestion
            setcookie('id', strval($id), (new DateTime('10 days'))->getTimestamp(), null, null, false, true); // 10 jours
```

WDYT ?

C'est un poil plus long à l'écriture, mais tu n'as pas besoin de commentaire pour comprendre que le cookie s'applique durant 10 jours.